test/e2e-tests/support/config.go

```diff
@@ -0,0 +1,101 @@
+package support
+
+import (
+	"encoding/json"
+	"os"
+)
+
+const (
+	S3CredentialsTemplatePath            = "../../../pipelines/tekton/aiedge-e2e/templates/credentials-s3.secret.yaml.template"
+	ImageRegistryCredentialsTemplatePath = "../../../pipelines/tekton/aiedge-e2e/templates/credentials-image-registry.secret.yaml.template"
+	GitCredentialsTemplatePath           = "../../../pipelines/tekton/gitops-update-pipeline/example-pipelineruns/example-git-credentials-secret.yaml.template"
+
+	ConfigPath = "../config.json"
+)
+
+var (
+	config *Config = nil
+)
+
+type Config struct {
+	Namespace             string   `json:"NAMESPACE"`
+	ImageRegistryUsername string   `json:"IMAGE_REGISTRY_USERNAME"`
+	ImageRegistryPassword string   `json:"IMAGE_REGISTRY_PASSWORD"`
+	TargetImageTags       []string `json:"TARGET_IMAGE_TAGS"`
+
+	GitFetchConfig GitFetchConfig `json:"git-fetch"`
+	S3FetchConfig  S3FetchConfig  `json:"s3-fetch"`
+	GitOpsConfig   GitOpsConfig   `json:"gitops"`
+
+	Clients         *Clients
+	GitFetchEnabled bool
+	S3FetchEnabled  bool
+	GitOpsEnabled   bool
```

Comment:
Just about what we were saying, moving these to the `GitFetchConfig ` etc. could work